Lib/test/test_urllib.py

```diff
@@ -1503,6 +1509,23 @@ def test_url2pathname(self):
         self.assertEqual(fn('//localhost/foo/bar'), f'{sep}foo{sep}bar')
         self.assertEqual(fn('///foo/bar'), f'{sep}foo{sep}bar')
         self.assertEqual(fn('////foo/bar'), f'{sep}{sep}foo{sep}bar')
+        self.assertEqual(fn('file:', has_scheme=True), '')
```

Comment:
let's also add tests with multiple schemes, e.g., `file://data://` (but that are not correct) and where we have a `data://` scheme for instance (is it possible to have a file named 'data://' actually?))

Reply:
>  is it possible to have a [path] named 'data://' actually?

Yes! Its URI form would be `file:data://`.